Array 1.39.0 release (#4056)
PostHog/posthog.com: contents/blog/the-posthog-array-1-39-0.mdx

---
date: 2022-08-29
title: 'Array 1.39.0: Moving person properties onto events'
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.39.0 moves person properties onto events, resulting in queries which are up to 400x faster. And that's just for starters!
---

Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!

> Running a self-hosted instance? Check out our [Upgrading PostHog guide](/docs/runbook/upgrading-posthog).

## PostHog 1.39.0 release notes

**Release highlights:**

- New: Joe to add Persons on Events here
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Apps: JOE to add

> ### New: Person properties have moved on to events

Author: @tiina303 @EDsCODE I've added some content in here. We'll link to the full post when it's done.

We've made substantial changes to the way [persons](/manual/persons) and [events](/manual/events) work by moving all persons properties _onto_ events. It's a substantial change, months in the planning, which is critical to PostHog's ability to scale. 

> A big benefit of this approach is that it enables significantly faster queries because we no longer have to compare multiple tables when running queries - everything is stored on the event. Some of our internal tests have shown a 400x increase in speed!

Reviewer: this feels misleading, we saw 400x increase for a single query, this makes me think it's across the fleet 400x

Author: Ah, gotcha. Had misunderstood this. 

```suggestion
A big benefit of this approach is that it enables significantly faster queries because we no longer have to compare multiple tables when running queries - everything is stored on the event. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 
```

This is a big change and one which we're planning to cover in more detail soon. Stay tuned to [the PostHog blog](/blog) for updates. 

### New: 

Paul to add a highlight here. Just one or two sentences is enough. 

### New: 

Paul to add a highlight here. Just one or two sentences is enough. 

### New: 

Paul to add a highlight here. Just one or two sentences is enough. 

### New: 

Paul to add a highlight here. Just one or two sentences is enough. 

> ### Improved: 

Reviewer: The PR is https://github.com/PostHog/posthog/pull/10912 

```suggestion
### Improved: Async migrations that aren't available yet will show up in a separate tab and migrations in Starting state can now be stopped in the UI.
```

Reviewer: @pauldambra

Author: Added

---
date: 2022-08-29
title: 'Array 1.39.0: Moving person properties onto events'
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.39.0 introduces a new beta option which could improve query speed by up to 400x. And that's just for starters!
---

Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!

> Running a self-hosted instance? Check out our [Upgrading PostHog guide](/docs/runbook/upgrading-posthog).

## PostHog 1.39.0 release notes

**Release highlights:**

- Beta: [Person properties on events](#beta-person-properties-have-moved-on-to-events)
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Apps: JOE to add

> ### Beta: Person properties on events

Reviewer: ```suggestion
### Beta: Person on events
```

> We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and moving all persons properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release by toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings).

Reviewer: We will still keep them the old way too + the person id part is important for uniq persons queries.
```suggestion
We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release by toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings).
```

Reviewer: 1.39.0 gives you the chance to try the feature ahead of the full release by toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings).

two conditions here: 1. async migration completed + toggle

> So, why have we moved person properties onto events? In part it's because it streamlines how we store data as we continue to scale, but it's also to unlock significant speed increases on queries involving both persons and events. Previously, PostHog stored personIDs and event information in separate tables, but this update simplifie

Reviewer: There were two reasons actually: 
1. scale
2. being able to query based on person properties at the event time (feel free to list this one as the first item)

sth like 

So, why are we making this change? There's two reasons: 
1. By talking to customers we realised how they want to query data and look at insights, but it wasn't possible in PostHog in the past - specifically it wasn't possible to query based on person properties at the time of the event. Now that we're putting person properties on the events this will be the default. Note that it's still possible to filter insights based on latest properties too - that can be done by using cohorts.
2. Scale - previously we needed to merge events distinct id and persons tables now we can just look up all the info in the events table itself and indexes and materialising columns helps us speed up queries even more.

A big benefit of this approach is that it enables significantly faster queries because we no longer have to compare multiple tables when running queries - everything is stored on the event. Some of our internal tests have shown a 400x increase in speed!

> We'll share more about this shift once it's fully released to all users - keep and eye on [the PostHog blog](/blog) for updates. 

Reviewer: I'd add something about how `filtering by unique users` and `filtering by person properties` will work differently once enabled - read more about how it all works and what changed + how to enable this on your self-hosted instance

Author: Yep, just waiting on a link from @pjhul before I add this in.

---
date: 2022-08-29
title: 'Array 1.39.0: Moving person properties onto events'
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.39.0 introduces a new beta option which could improve query speed by up to 400x. And that's just for starters!
---

Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!

> Running a self-hosted instance? Check out our [Upgrading PostHog guide](/docs/runbook/upgrading-posthog).

## PostHog 1.39.0 release notes

**Release highlights:**

- Beta: [Persons on events](#beta-persons-on-events)
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- New: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Improved: Paul to add a highlight here
- Apps: JOE to add

### Beta: Persons on events

> We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release by toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings).

Author: ```suggestion
We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release by [running an async migration](/docs/runbook/async-migrations) and toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings). 
```

We'll share more about this update once it's fully released to all users - keep and eye on [the PostHog blog](/blog) for updates. 

### New: Format your insights to show durations

![formatting an insight as a duration](../images/blog/array-1.39.0/duration-formatting.gif)

Do your event properties contain milliseconds or seconds? Then you can now make things easier by changing your Y-axis unit to show durations instead of the default `none`. It's a small addition, but one which makes insights much easier to read and share with others.

### New: Big, bold number chart type

![the big bold number insight formatted as a duration](../images/blog/array-1.39.0/big-number.png)

This new, much requested and _very_ exciting big bold number insight type does exactly what it says: simplifies a query result down to a single number. And it works great with durations too!

### Improved: Turbo properties

Frustrated that it was taking a few seconds to load properties? You weren't the only one. 

> Thankfully, version 1.39.0 addresses this by adding a limit so that PostHog doesn't always search through hundreds of thousands of properties for each query. The result? [Properties now load much faster](https://github.com/PostHog/posthog/pull/11037). 

Reviewer 2: ```suggestion
Thankfully, version 1.39.0 addresses this by adding a limit so that PostHog doesn't always search through hundreds of thousands of properties for each query. The result? [Properties now load much faster] On PostHog Cloud we saw a 75% reduction in API call time. (https://github.com/PostHog/posthog/pull/11037). 
```

Reviewer 2: @joethreepwood worth including the improvement?

### Apps: Census integration

Census, an operational analytics platform for syncing data with tools across your stack, has created a new PostHog integration so you can sync events data into their platform. Check [the Census docs](https://docs.getcensus.com/destinations/posthog) on how to get started!

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Other improvements & fixes
Version 1.39 also adds hundreds of other improvements and fixes, including...

-   **Improvement**: We've removed some axe violations on Home, to improve accessibility. And added accessibility testing to our workflow. Bonus: Check the PR for [a _fantastic_ Dad joke](https://github.com/PostHog/posthog/pull/11211#discussion_r941651022). 
-   **Improvement**: When deleting persons, you can now choose to delete event data as well thanks to [a new person deletion flow](https://github.com/PostHog/posthog/pull/11347).

> -   **Improvement**: We've [optimised the query behind loading property definitions](https://github.com/PostHog/posthog/pull/11037). For PostHog Cloud we saw a 75% reduction in API call time.

Reviewer 2: ```suggestion
```

this is in twice

- [Improved: Turbo properties](#improved-turbo-properties)
- [Improved: History log now shows changes to insights](#improved-history-log-now-shows-changes-to-insights)
- [Apps: Census integration](#apps-census-integration)

### Beta: Persons on events

We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release. Here's how:

- **PostHog Cloud:** No action is needed, as we're rolling this test out steadily behind a feature flag. If you're eager to be included from the start, let us know in [Slack](/slack) to see about joining our beta group.
-  **Self-hosted users:** First, you need to [run an async migration](/docs/runbook/async-migrations). Afterwards, change your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.

> Why are we doing this? Feedback showed that users weren't able to create queries based on person properties at the time of an event. By putting person properties on events, this becomes the new default, while still enabling you to to filter insights based on the latest properties using cohorts. 

Reviewer 3: This is not the important thing here. Feedback from users has shown that query performance is paramount.

featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.39.0 introduces a new beta for you to try, a new display chart and big improvements to our client libraries.
---

Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!

> Running a self-hosted instance? Check out our [guide to upgrading PostHog](/docs/runbook/upgrading-posthog).

## PostHog 1.39.0 release notes

**Release highlights:**

- Beta: [Persons on events](#beta-persons-on-events)
- [New: Format your insights to show durations](#new-format-your-insights-to-show-durations)
- [New: Big, bold number chart type](#new-big-bold-number-chart-type)
- [Improved: Turbo properties](#improved-turbo-properties)
- [Improved: History log now shows changes to insights](#improved-history-log-now-shows-changes-to-insights)
- [Improved: Client and server-side library improvements](#improved-client-and-server-side-library-improvements)
- [Improved: Automatically comment SQL queries](#improved-automatically-comment-sql-queries)
- [Apps: Census integration](#apps-census-integration)

> ### Beta: Persons on events

Reviewer 4: Is this too technical for a title? Should we do something funnier/more exciting?

Reviewer 5: I second this. I think for most readers who have no context on this change or on how our data is structured this won't stand out as much as it should.

Author: Open to suggestions here, as it's a bit of a complex thing to parse anyway (persons not people), but I've tweaked this to make the benefit clearer to users: `Beta: Improving query performance by combining persons and events`

We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release. Here's how:

- **PostHog Cloud:** No action is needed, as we're rolling this test out steadily behind a feature flag. If you're eager to be included from the start, let us know in [Slack](/slack) to see about joining our beta group.
-  **Self-hosted users:** First, you need to [run an async migration](/docs/runbook/async-migrations). Afterwards, change your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.

Why are we doing this? Feedback showed that users weren't able to create queries based on person properties at the time of an event. By putting person properties on events, this becomes the new default, while still enabling you to to filter insights based on the latest properties using cohorts. 

> Also, it makes queries significantly faster because we no longer have to merge tables to get a result; we can just look up everything in the events table instead. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 

Reviewer 3: Flip the sections. Querying speed is the most important win here, everything else is salad dressing and should be deemphesized.

There'll be some slight changes to how filtering by unique users or person properties work as a result of this change. For more information on this and to find out how we've implemented it change behind the scenes, check out our [overview of the ingestion pipeline](/docs/architecture/ingestion-pipeline).

### New: Format your insights to show durations

![formatting an insight as a duration](../images/blog/array-1.39.0/duration-formatting.gif)

Do your event properties contain milliseconds or seconds? Then you can now make things easier by changing your Y-axis unit to show durations instead of the default `none`. It's a small addition, but one which makes insights much easier to read and share with others.

### New: Big, bold number chart type

![the big bold number insight formatted as a duration](../images/blog/array-1.39.0/big-number.png)

This new, much requested and _very_ exciting big bold number insight type does exactly what it says: simplifies a query result down to a single number. And it works great with durations too!

### Improved: Turbo properties

Frustrated that it was taking a few seconds to load properties? You weren't the only one. 

> Thankfully, version 1.39.0 addresses this by adding a limit so that PostHog doesn't always search through hundreds of thousands of properties for each query. The result? [Properties now load much faster] On PostHog Cloud we saw a 75% reduction in API call time. (https://github.com/PostHog/posthog/pull/11037). 

Reviewer 2: ```suggestion
Thankfully, version 1.39.0 addresses this by adding a limit so that PostHog doesn't always search through hundreds of thousands of properties for each query. The result? [Properties now load much faster](https://github.com/PostHog/posthog/pull/11037). On PostHog Cloud we saw a 75% reduction in API call time. 
```

### Improved: History log now shows changes to insights

![a view of the insights activity tab](../images/blog/array-1.39.0/insights-activity.png)

We've had feedback from several users that they enjoy tracking changes to their feature flags in the history logs. So, we added the same functionality to insights too! Now you can snoop on who has been changing your insights and giving them funny names.

> ## Improved: Client and server-side library improvements

Reviewer 4: This should be way higher, probably the most/second most important thing that's happened this release?

Reviewer 4: @neilkakkar @mariusandra anything to add onto this?

Author: Moved to second position and gave it a better title

Reviewer 6: I looked through the past arrays, and it seems we haven't announced anywhere the [expo-compatible](https://expo.dev/) React Native library v2 rewrite. It opens us up to a huge new market (vs the non expo-compatible v1), and is probably worth announcing to users in the array.

Author: Will add to this section!

While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. We've also updated our server-side libraries to have local feature flag evaluation.

Check [our library docs](/docs/integrate/libraries) to find out more about what each library supports and how to deploy. 

> ### Improved: Automatically comment SQL queries

Reviewer 4: I'd remove this, too internal

Reviewer 3: This is worth highlighting under 'other improvements' as clients who we've helped debug things have asked/needed this before.

### Apps: Census integration

Census, a reverse ETL platform which syncs your data warehouse with your operational tools, has created a new PostHog integration. Now you can easily sync events data such as sales or marketing activity into PostHog. Check [the Census docs](https://docs.getcensus.com/destinations/posthog) to find out more!

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Other improvements & fixes
Version 1.39 also adds hundreds of other improvements and fixes, including...

- **Improvement**: Async migrations that aren't available yet will show up in a separate tab and migrations in Starting state can now be stopped in the UI. This solves [a few problems](https://github.com/PostHog/posthog/pull/10912).
-   **Improvement**: We've removed some axe violations on Home, to improve accessibility. And added accessibility testing to our workflow. Bonus: Check the PR for [a _fantastic_ Dad joke](https://github.com/PostHog/posthog/pull/11211#discussion_r941651022). 

> -   **Improvement**: When deleting persons, you can now choose to delete event data as well thanks to [a new person deletion flow](https://github.com/PostHog/posthog/pull/11347).

Reviewer 4: This should maybe be higher/it's own section? GDPR compliance etc

Reviewer 3: +1

-   **Improvement**: For users self-hosting PostHog, you can now set-up [monitoring and logging through Grafana](/docs/self-host/configure/monitoring-with-grafana) out of the box, which makes it much easier to monitor and troubleshoot your deployment.
-   **Improvement**: [We've separated the list of feature flags on an event from its event properties in the taxonomic filter](https://github.com/PostHog/posthog/pull/10668). This makes it (approximately) 1,000,000% easier to find things when there are lots of feature flags.

> - **Improvement**: We've introduced [the snappy compression codec](https://google.github.io/snappy/) for compressing in-flight messages. No more 'message too large' errors!

Reviewer 4: Too internal?

Reviewer 3: ```suggestion
- **Improvement**: We've introduced [the snappy compression codec](https://google.github.io/snappy/) for compressing in-flight messages, which reduces some kafka topic sized by up to 4x. No more 'message too large' errors!
```

Author: ```suggestion
- **Improvement**: We've introduced [the snappy compression codec](https://google.github.io/snappy/) for compressing in-flight messages, which reduces some kafka topic sizes by up to 4x. No more 'message too large' errors!
```

-   **Fix**: In a few unexpected cases we used to return a successful response to some invalid API requests to the capture endpoint. [Now we correctly return a `400 BAD REQUEST`](https://github.com/PostHog/posthog/pull/10999) which will help people with useful feedback during development

> View the commit log in GitHub for a full history of changes: [`release-1.37.0...release-1.38.0`](https://github.com/PostHog/posthog/compare/release-1.38.0...release-1.39.0).

Reviewer 2: ```suggestion
View the commit log in GitHub for a full history of changes: [`release-1.38.0...release-1.39.0`](https://github.com/PostHog/posthog/compare/release-1.38.0...release-1.39.0).
```

- [New: Added features for client and server-side libraries](#new-added-features-for-client-and-server-side-libraries)
- [New: Format your insights to show durations](#new-format-your-insights-to-show-durations)
- [New: Big, bold number chart type](#new-big-bold-number-chart-type)
- [Improved: Turbo properties](#improved-turbo-properties)
- [Improved: History log now shows changes to insights](#improved-history-log-now-shows-changes-to-insights)
- [Apps: Census integration](#apps-census-integration)

### Beta: Improving query performance by combining persons and events

We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release. Here's how:

- **PostHog Cloud:** No action is needed, as we're rolling this test out steadily behind a feature flag. If you're eager to be included from the start, let us know in [Slack](/slack) to see about joining our beta group.

> -  **Self-hosted users:** First, you need to [run an async migration](/docs/runbook/async-migrations). Afterwards, change your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.

Reviewer 7: I think we should be specific here?

```suggestion
-  **Self-hosted users:** First, you need to [run async migration number 0006](/docs/runbook/async-migrations). Afterwards, change your `PERSON_ON_EVENTS_ENABLED` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.
```

Why are we doing this? Firstly, it makes queries significantly faster because we no longer have to merge tables to get a result; we can just look up everything in the events table instead. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 

Secondly, feedback showed that users weren't able to create queries based on person properties at the time of an event. By putting person properties on events, this becomes the new default, while still enabling you to to filter insights based on the latest properties using cohorts. 

There'll be some slight changes to how filtering by unique users or person properties work as a result of this change. For more information on this and to find out how we've implemented it change behind the scenes, check out our [overview of the ingestion pipeline](/docs/architecture/ingestion-pipeline).

## New: Added features for client and server-side libraries

> While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. We've also updated our server-side libraries to have local feature flag evaluation.

Reviewer 7: The server-side libraries are a pretty huge deal, so my attempt to highlight this a bit more:

```suggestion
While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. 

Our server-side libraries now have local feature flag evaluation. This is huge because now you can use feature flags in your backend without worrying about latency. Evaluating locally makes the computation time go down from an average of 400ms to 10ms.
```

Author: > The server-side libraries are a pretty huge deal, so my attempt to highlight this a bit more:

```suggestion
While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. 

We've also updated our server-side libraries with local feature flag evaluation. This is huge because you can now use feature flags in your backend without worrying about latency. Evaluating locally reduces the computation time from an average of 400ms to 10ms!
```

- [Improved: Turbo properties](#improved-turbo-properties)
- [Improved: History log now shows changes to insights](#improved-history-log-now-shows-changes-to-insights)
- [Apps: Census integration](#apps-census-integration)

### Beta: Improving query performance by combining persons and events

We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release. Here's how:

- **PostHog Cloud:** No action is needed, as we're rolling this test out steadily behind a feature flag. If you're eager to be included from the start, let us know in [Slack](/slack) to see about joining our beta group.
-  **Self-hosted users:** First, you need to [run async migration number 0006](/docs/runbook/async-migrations). Afterwards, change your `PERSON_ON_EVENTS_ENABLED` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.

> Why are we doing this? Firstly, it makes queries significantly faster because we no longer have to merge tables to get a result; we can just look up everything in the events table instead. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 

Reviewer: nit: 
```suggestion
Why are we doing this? Firstly, it makes queries significantly faster because we no longer have to join tables to get a result; we can just look up everything in the events table instead. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 
```

Secondly, feedback showed that users weren't able to create queries based on person properties at the time of an event. By putting person properties on events, this becomes the new default, while still enabling you to to filter insights based on the latest properties using cohorts. 

> There'll be some slight changes to how filtering by unique users or person properties work as a result of this change. For more information on this and to find out how we've implemented it change behind the scenes, check out our [overview of the ingestion pipeline](/docs/architecture/ingestion-pipeline).

Reviewer: ?
```suggestion
There'll be some slight changes to how filtering by unique users or person properties work as a result of this change. For more information on this and to find out how we've implemented the change behind the scenes, check out our [overview of the ingestion pipeline](/docs/architecture/ingestion-pipeline).
```

## New: Added features for client and server-side libraries

While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. 

> We've also updated our server-side libraries with local feature flag evaluation. This is huge because you can now use feature flags in your backend without worrying about latency. Evaluating locally reduces the computation time from an average of 400ms to 10ms!

Author: ```suggestion
We've also updated our server-side libraries with local feature flag evaluation. This is huge because you can now use feature flags in your backend without worrying about latency. Evaluating locally reduces the computation time from an average of 400ms to 10ms!

Also, in case you missed it previously, we've rewritten the React Native library, bringing it up to a 2.0 which is [expo-compatible](https://expo.dev/). We actually did this a little while ago and didn't announce it at the time, which is why you may have missed it before. Sorry!
```

@mariusandra this cover it?

Reviewer 7: It was released on 2nd August, so this array is the right one: https://github.com/PostHog/posthog-js-lite/pull/10

No need for the last line :)

Reviewer 7: oh, maybe worth covering that the minimum PostHog version requirement for all these new libraries is 1.38 (we mention it in the CHANGELOG, but would be nice to highlight here as well for self hosted peeps planning to upgrade libraries after reading this)

Reviewer 6: @benjackwhite is the right person to 👍 / 👎 this

Reviewer 6: I forgot that he's off today. I think the RN blob requires its own bullet point / header, as these big things got released all at once:
- React Native integration fully rewritten in TS. This means no external iOS/Android library dependencies and thus it now works with Expo.
- Added support for magic autocapture (that's a new for the react native library - only custom instrumented events were supported before, now we do magic)
- Added support for feature flags and group analytics (most libraries got these recently)

Author: Added a commit to reflect this and will add to the email now too. Thanks!

This new, much requested and _very_ exciting big bold number insight type does exactly what it says: simplifies a query result down to a single number. And it works great with durations too!

> ## Improved: New person deltion flow

Reviewer 3: ```suggestion
## Improved: New person deletion flow
```
